rc/base/markdown.kak

```diff
@@ -12,51 +12,52 @@ hook global BufCreate .*[.](markdown|md|mkd) %{
-    tupfile    ```\h*tupfile    ```          '' \
-    yaml       ```\h*yaml       ```          '' \
-    code       ```              ```          '' \
-    code       ``               ``           '' \
-    code       `                `            ''
+    c          ^```\h*c\b\K          ```          '' \
+    cabal      ^```\h*cabal\b\K      ```          '' \
+    clojure    ^```\h*clojure\b\K    ```          '' \
+    coffee     ^```\h*coffee\b\K     ```          '' \
+    cpp        ^```\h*cpp\b\K        ```          '' \
+    css        ^```\h*css\b\K        ```          '' \
+    cucumber   ^```\h*cucumber\b\K   ```          '' \
+    d          ^```\h*d\b\K          ```          '' \
+    diff       ^```\h*diff\b\K       ```          '' \
+    dockerfile ^```\h*dockerfile\b\K ```          '' \
+    fish       ^```\h*fish\b\K       ```          '' \
+    gas        ^```\h*gas\b\K        ```          '' \
+    go         ^```\h*go\b\K         ```          '' \
+    haml       ^```\h*haml\b\K       ```          '' \
+    haskell    ^```\h*haskell\b\K    ```          '' \
+    html       ^```\h*html\b\K       ```          '' \
+    ini        ^```\h*ini\b\K        ```          '' \
+    java       ^```\h*java\b\K       ```          '' \
+    javascript ^```\h*javascript\b\K ```          '' \
+    json       ^```\h*json\b\K       ```          '' \
+    julia      ^```\h*julia\b\K      ```          '' \
+    kak        ^```\h*kak\b\K        ```          '' \
+    kickstart  ^```\h*kickstart\b\K  ```          '' \
+    latex      ^```\h*latex\b\K      ```          '' \
+    lisp       ^```\h*lisp\b\K       ```          '' \
+    lua        ^```\h*lua\b\K        ```          '' \
+    makefile   ^```\h*makefile\b\K   ```          '' \
+    moon       ^```\h*moon\b\K       ```          '' \
+    objc       ^```\h*objc\b\K       ```          '' \
+    perl       ^```\h*perl\b\K       ```          '' \
+    pug        ^```\h*pug\b\K        ```          '' \
+    python     ^```\h*python\b\K     ```          '' \
+    ragel      ^```\h*ragel\b\K      ```          '' \
+    ruby       ^```\h*ruby\b\K       ```          '' \
+    rust       ^```\h*rust\b\K       ```          '' \
+    sass       ^```\h*sass\b\K       ```          '' \
+    scala      ^```\h*scala\b\K      ```          '' \
+    scss       ^```\h*scss\b\K       ```          '' \
+    sh         ^```\h*sh\b\K         ```          '' \
+    swift      ^```\h*swift\b\K      ```          '' \
+    tupfile    ^```\h*tupfile\b\K    ```          '' \
+    typescript ^```\h*typescript\b\K ```          '' \
+    yaml       ^```\h*yaml\b\K       ```          '' \
+    code       ^```((!?=(c|cabal|clojure|coffee|cpp|css|cucumber|diff|dockerfile|fish|gas|go|haml|haskell|html|ini|java|javascript|json|julia|kakrc|kickstart|latex|lisp|lua|makefile|moon|objc|perl|pug|python|ragel|ruby|rust|sass|scala|scss|sh|swift|tupfile|typescript|yaml)[^\n])*)$          ```          '' \
```

Comment:
hum this is fishy, `!?=` is not negative lookahead...

|            | lookahead | lookbehind
------------ | --------- | ----------
**positive** | `?=`      | `?<=`
**negative** | `?!`      | `?<!`

Reply:
Good point, maybe that regex is just failing and thus doesn't override the previous blocks. I tried to modify the `code` regex, but I get the error:

```
markdown.kak:14:1: 'add-highlighter' regex parse error: Lookaround can only contain literals, any chars or character classes at '^```(?!(c|cabal|clojure|coffee|cpp|css|cucumber|diff|dockerfile|fish|gas|go|haml|haskell|html|ini|java|javascript|json|julia|kakrc|kickstart|latex|lisp|lua|makefile|moon|objc|perl|pug|python|ragel|ruby|rust|sass|scala|scss|sh|swift|tupfile|typescript|yaml))<<<HERE>>>[^\n]*$'
```

So maybe you can't use `|` in the lookahead.

Reply:
This seems to fix it:

```
code       ^```(c|cabal|clojure|coffee|cpp|css|cucumber|diff|dockerfile|fish|gas|go|haml|haskell|html|ini|java|javascript|json|julia|kakrc|kickstart|latex|lisp|lua|makefile|moon|objc|perl|pug|python|ragel|ruby|rust|sass|scala|scss|sh|swift|tupfile|typescript|yaml)\K          ```          '' \
```

I'll make a PR with that.